### PR TITLE
Add support for PHP 8.2, drop support for PHP 7.4

### LINKED ISSUE #103: Support for PHP 8.2
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no
| BC Break    | no

#### Summary

I'm creating this as I don't see a ticket to add support for php 8.2

I'm willing to contribute the change if needed.

### PR DESCRIPTION
**Pull request by ghostwriter**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

- Update composer config `platform.php` to `8.0.99`
- Update composer `PHP` to `~8.0.0 || ~8.1.0 || ~8.2.0`
- Ignore PHP platform requirements via `.laminas-ci.json`
- Resolves #103

### CODE REVIEW

**Comment by Ocramius** [comment]
@weierophinney we probably need a new build of swoole here 😬

**Comment by weierophinney** [comment]
> we probably need a new build of swoole here

Will get it built later today!

**Comment by weierophinney** [comment]
Support is ready, and I've created a PR for laminas-continuous-integration-action (laminas/laminas-continuous-integration-action#139) to allow using the new packages. When that's released, we can rerun the PHP 8.2 jobs.

**Comment by Ocramius** [comment]
Iz still building: https://github.com/laminas/laminas-continuous-integration-action/actions/runs/3481802570/jobs/5823355178

**Comment by ghostwriter** [comment]
> E: Unable to locate package php8.2-inotify

**Comment by Ocramius** [comment]
Where do we install inotify? Manually?

**Comment by weierophinney** [comment]
> Where do we install inotify? Manually?

Every other PHP version we install on has this pre-built; it's why we have it i the `.laminas-ci.json` file. However, for 8.2... Ondrej has not included it (yet).

We may have to build it in the CI container. :slightly_frowning_face:

**Comment by ghostwriter** [comment]
We tell the ci to install it, via `.laminas-ci.json` https://github.com/mezzio/mezzio-swoole/blob/09b1d45b7b6b4258ecf8aa3e7c88db3b0f257991/.laminas-ci.json#L3

testing locally…

`docker run -v $(pwd):/github/workspace -w=/github/workspace ghcr.io/laminas/laminas-continuous-integration:1.30 '{"command":"./vendor/bin/phpunit","php":"8.2","extensions":["inotify","swoole"],"ini":[],"dependencies":"locked"}'`

**Comment by ghostwriter** [comment]
Successfully installed packaged extensions: `php8.2-inotify`,

Ready for review.

**Comment by Ocramius** [comment]
Could this be rebased, rather than merged?

File: src/StaticResourceHandler/FileLocationRepository.php [hunk 1]
```diff
@@ -100,8 +100,13 @@ private function normalizeDirectory(string $directory, string $prefix): string
...
         }
-
-        if ($directory === '/' || $directory === '\\' || $directory === '\\\\') {
+        if ($directory === '/') {
+            return '/';
+        }
+        if ($directory === '\\') {
+            return '/';
+        }
+        if ($directory === '\\\\') {
```
Comment (Ocramius) [inline src/StaticResourceHandler/FileLocationRepository.php:109]: 'bit of duplication, but it seems like this code was changed via automation

**Review by Ocramius — APPROVED** [review]
Excellent, thanks @ghostwriter!

**Pull request merged**